Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -777,22 +777,16 @@ private bool IsProjectGlobalDeclaration(Declaration candidateDeclaration, Declar
         {
             return candidateDeclaration.ProjectName == scopingDeclaration.ProjectName
                 && !(candidateDeclaration.ParentScopeDeclaration is ClassModuleDeclaration)
-                && (IsExplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration)
-                || IsImplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration));
+                && (IsPublicInOtherModule(candidateDeclaration, scopingDeclaration));
         }
 
-        private bool IsExplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
+        private bool IsPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
         {
             return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
-                    && candidateDeclaration.Accessibility == Accessibility.Public;
-        }
-
-        private bool IsImplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
-        {
-            return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
-                        && candidateDeclaration.Accessibility == Accessibility.Implicit
-                        && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)
-                        && !candidateDeclaration.IdentifierName.StartsWith("Option ");
+                        && (candidateDeclaration.Accessibility == Accessibility.Public
+                            || (candidateDeclaration.Accessibility == Accessibility.Implicit)
+                                && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)
```

Comment:
In this PR, the reason to find all accessible declarations is to generate a list of names that would cause a collision (or simply be a poor choice) in a renaming operation.  However, I think I am running up against slightly different expectations of the term "accessible".  It appears as though, from a renaming perspective, the result for calls to AccessibilityCheck.IsAccessible(...) is more restrictive in one sense and more generous in another.  

**The more restrictive case:**
`Public Sub TheFunctionName()
	Dim TheVariableName as Long
          ....
End Sub'

Renaming "TheFunctionName" to "TheVariableName" would be a poor choice.  "TheVariableName" however, is considered inaccessible by the AccessibilityCheck criteria.  And so, by using accessibility criteria exclusively, would be an acceptable rename. 

**The more generous case:**
Given two ClassModules "TheFirstClass" and "TheSecondClass"
Within class "TheFirstClass" any Public Sub, Function, or member variable of the "TheSecondClass" is considered accessible by the AccessibilityCheck criteria.  For renaming purposes though, it seems to me that the user should be able to rename a variable or function within "TheFirstClass" to have the same identifier as any of "TheSecondClass" Public identifiers free of any kind of warning.

In the more generous case(s), additional filtering can be done to the AccessibilityCheck results.  For the more restrictive case, it seems that I may need to re-filter the AllUserDeclarations list for this special case.  Also, I'm wondering if the functions and user messages for Renaming operations should be using terminology more like "collision" or "confusion" rather than "accessibility".  Thoughts/Reactions?